CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -732,6 +733,16 @@ namespace CefSharp
         bool ClientAdapter::GetAuthCredentials(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, bool isProxy,
             const CefString& host, int port, const CefString& realm, const CefString& scheme, CefRefPtr<CefAuthCallback> callback)
         {
+            if (isProxy && Cef::proxyOptions != nullptr && Cef::proxyOptions->IP == StringUtils::ToClr(host) && !String::IsNullOrEmpty(Cef::proxyOptions->Username)
```

Comment:
What if you wanted to provide a different `username` and `password` after you've initialized `CEF`? 

I'd rather not having additional code here, specially that adds a new behavior. 

It probably makes more sense to provide a default implementation of `IRequestHandler` that contains this logic. (If we do go down that path, probably should add default implementations for all the handlers.)